test/lib/hermione.js

```
@@ -157,6 +158,73 @@ describe('hermione', () => {
                 return hermione.run()
                     .then(() => assert.callOrder(afterInit, Runner.prototype.run));
             });
+
+            it('should send INIT event only once', () => {
+                const onInit = sinon.spy();
+                const hermione = Hermione.create()
+                    .on(RunnerEvents.INIT, onInit);
+
+                return hermione.run().then(hermione.run()).then(() => {
+                    assert.calledOnce(onInit);
+                });
+            });
+        });
+
+        describe('reporters', () => {
+            let HermioneQuire;
+            let attachRunner;
+            let runner;
+
+            const createReporter = () => {
+                return function Reporter() {
+                    this.attachRunner = attachRunner;
+                };
+            };
+
+            beforeEach(() => {
+                HermioneQuire = proxyquire('lib/hermione', {
+                    './reporters/reporter': createReporter()
+                });
+
+                runner = mkRunnerStub_();
+                attachRunner = sandbox.stub();
+            });
+
+            it('should accept reporter as string', () => {
+                const reporter = 'reporter';
+                const options = {reporters: [reporter]};
+
+                return HermioneQuire.create().run(null, options).then(() => {
+                    assert.calledOnceWith(attachRunner, runner);
+                });
+            });
+
+            it('should throw if reporter is not found', () => {
```

should throw if reporter **was** not found for given identifier